### ISSUE TITLE
Last Finalized Block does not update when upgrading from v6 to v6.1

### ISSUE THREAD

**Issue by wpank** [paritytech/polkadot#463]
After running a Kusama node on `v0.6` I upgrade to `v0.6.1` by running

```
cargo clean
git checkout v0.6.1
./scripts/init.sh
cargo build --release
```
I then ran the node `./target/release/polkadot`

And the `Block` will sync to the latest, while the last `Finalized Block` will remain stuck at 100,000 blocks behind. 

The terminal output will warn the following:
```
2019-10-06 18:15:06 Safety violation: attempted to revert finalized block HashAndNumber { number: 50806, hash: 0xe95778eef63b697351af3d366057f5967ad8b49a9749a022a6b3e495dbb9c0f4 } which is not in the same chain as last finalized 0xe95778eef63b697351af3d366057f5967ad8b49a9749a022a6b3e495dbb9c0f4
2019-10-06 18:15:06 Error applying finality to block (0xb34fe9e2ae8d4b8bd692d5d275998764f21eca73ca43dc3a53b981e31178fb60, 145432): NotInFinalizedChain
2019-10-06 18:15:06 GRANDPA Observer failed: Client(NotInFinalizedChain)
```

I've noticed a few other nodes on Telemetry that are also stuck behind in Finalized Blocks, many of them on `v0.6.1`. Unsure if this is an issue because of upgrading the node or something else.

**Comment by andresilva** [comment]
Thanks for the report. Fixed in https://github.com/paritytech/substrate/pull/3776. Will be included in v0.6.2 which will be released shortly.

**Comment by wpank** [comment]
@andresilva Thanks! Upgraded to `v0.6.2` and can confirm this was fixed for my node. Closing this~
